sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java

```diff
@@ -0,0 +1,400 @@
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+  })
+  @GET
+  @Path("{document-id}/{document-path:.*}")
+  public Uni<RestResponse<Object>> getDocumentPath(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @PathParam("document-id") String id,
+      @PathParam("document-path") List<PathSegment> path,
+      @QueryParam("where") String where,
+      @QueryParam("fields") String fields,
+      @QueryParam("page-size")
+          @Min(value = 1, message = "the minimum number of results to return is one")
+          Integer pageSize,
+      @QueryParam("page-state") String pageState,
+      @QueryParam("profile") boolean profile,
+      @QueryParam("raw") boolean raw) {
+
+    // fetch a valid table to ensure read is from a collection table
+    return tableManager
+        .getValidCollectionTable(namespace, collection)
+
+        // if exists, then map to the read action
+        .flatMap(
+            t -> {
+              boolean isSearch = where != null || pageSize != null;
+              ExecutionContext context = ExecutionContext.create(profile);
+
+              List<String> pathStrings = path.stream().map(PathSegment::getPath).toList();
+
+              if (isSearch) {
+                int pageSizeFinal = Optional.ofNullable(pageSize).orElse(100);
+                Paginator paginator = new Paginator(pageState, pageSizeFinal);
+
+                return readDocumentsService
+                    .findSubDocuments(
+                        namespace, collection, id, pathStrings, where, fields, paginator, context)
+                    .map(
+                        result -> {
+                          if (null != result) {
+                            return rawHandler(raw).apply(result);
+                          } else {
+                            return RestResponse.noContent();
+                          }
+                        });
+              } else {
+                return readDocumentsService
+                    .getDocument(namespace, collection, id, pathStrings, fields, context)
+                    .map(
+                        result -> {
+                          if (null != result) {
+                            return rawHandler(raw).apply(result);
+                          } else {
+                            int code = Response.Status.NOT_FOUND.getStatusCode();
+                            String msg = "A document with the id %s does not exist.".formatted(id);
+                            ApiError error = new ApiError(msg, 404);
+                            return RestResponse.ResponseBuilder.create(code).entity(error).build();
+                          }
+                        });
+              }
+            });
+  }
+
+  private Function<DocumentResponseWrapper<? extends JsonNode>, RestResponse<Object>> rawHandler(
```

Comment:
Was curious about `? extends JsonNode` here: I am guessing that simple `Function<DRW<JsonNode>>` would work well unless there are places where one really wants `Function<DRW<ObjectNode>>` or such?
Might reduce need for casting too.

Reply:
Yea - as far as current usage, I don't see the need for `? extends JsonNode` - also I think in theory anything consuming the DocumentResponseWrapper doesn't need a more specific type than JsonNode

I changed this and will merge when CI passes!